10.0.0 Release (#4186)
cypress-io/cypress-documentation: content/guides/references/configuration.md

| Option                 | Default                           | Description                                                                                                                                                                                  |
| ---------------------- | --------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `baseUrl`              | `null`                            | URL used as prefix for [`cy.visit()`](/api/commands/visit) or [`cy.request()`](/api/commands/request) command's URL.                                                                         |

Reviewer: This needs updated to be a e2e specific configuration option. Setting at the root level will now throw an error.

Author: @jaffrepaul it looks like this might have ben missed as part of your work on https://github.com/cypress-io/cypress-documentation/pull/4350. Can you take a look?

Reviewer 2: @cowboy Without looking thru a proper diff, there could still be some necessary changes in https://github.com/cypress-io/cypress-documentation/pull/4280 which has been around for awhile. Can you approve so we can merge and discuss an audit and the current state of affairs? We've had to refactor and update a number of things along the way so an audit is certainly a good idea.

- Execute code in the
  [pluginsFile](/guides/references/configuration#Folders-Files).

<DocsImage src="/img/guides/test-runner-settings-nodejs-version.jpg" alt="Node version in Settings in Cypress App" ></DocsImage>

### Experiments

Configuration might include experimental options currently being tested. See
[Experiments](/guides/references/experiments) page.

## Testing Type-Specific Options

You can provide configuration options for either E2E or Component Testing by
creating `e2e` and `component` objects inside your Cypress configuration.

### e2e

These options are available to be specified inside the `e2e` configuration
object:

| Option               | Default                               | Description                                                                                                                                                                                         |
| -------------------- | ------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `setupNodeEvents`    | `null`                                | Function in which node events can be registered and config can be modified. Takes the place of the (deprecated) plugins file. [Please read the notes for examples on using this.](#setupNodeEvents) |

Reviewer 3: [a7bf6c4](https://github.com/cypress-io/cypress-documentation/commit/a7bf6c451c6f70423bc5cb19d34e16f84f8d8917)

> | `supportFile`        | `cypress/support/e2e.{js,jsx,ts,tsx}` | Path to file to load before test files load. This file is compiled and bundled. (Pass `false` to disable)                                                                                           |

Reviewer 3: #3808e83

> | `supportFile`        | `cypress/support/e2e.{js,jsx,ts,tsx}` | Path to file to load before test files load. This file is compiled and bundled. (Pass `false` to disable)                                                                                           |

Reviewer 3: [3808e83](https://github.com/cypress-io/cypress-documentation/commit/3808e830c210da59a19460776ff0d00bd7917939)

> | `specPattern`        | `cypress/e2e/**/*.cy.{js,jsx,ts,tsx}` | A String or Array of glob patterns of the test files to load.                                                                                                                                       |

Reviewer: It was noted that we are still maintaining the terminology `spec` for a file with tests
```suggestion
| `specPattern`        | `cypress/e2e/**/*.cy.{js,jsx,ts,tsx}` | A String or Array of glob patterns of the spec files to load.                                                                                                                                       |
```

> | `excludeSpecPattern` | `*.hot-update.js`                     | A String or Array of glob patterns used to ignore test files that would otherwise be shown in your list of tests. [Please read the notes on using this.](#excludeSpecPattern)                       |

Reviewer: It was noted that we are still maintaining the terminology `spec` for a file with tests
```suggestion
| `excludeSpecPattern` | `*.hot-update.js`                     | A String or Array of glob patterns used to ignore spec files that would otherwise be shown in your list of tests. [Please read the notes on using this.](#excludeSpecPattern)                       |
```

Author: We've possibly missed many things. You may want to consider opening a separate PR if you're going to make a bunch of changes!

Reviewer: I was just wanting to review! 😅

Reviewer: @cowboy & I sync'd offline. We decided it'd be best to leave this branch as the "Source of truth" for unification documentation. I'm going to open a Unification Audit PR to provide feedback so we can apply changes & verify the tests & deployments are still valid.

  }
}
```

:::

### component

These options are available to be specified inside the `component` configuration
object:

| Option               | Default                                  | Description                                                                                                                                                                                         |
| -------------------- | ---------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `devServer`          | `null`                                   | Required function used to configure the component testing dev server. [Please read the notes for examples on using this.](#devServer-devServerConfig)                                               |
| `devServerConfig`    | `null`                                   | Optional dev server configuration. Config options will be determined by the dev server. [Please read the notes for examples on using this.](#devServer-devServerConfig)                             |

> | `setupNodeEvents`    | `null`                                   | Function in which node events can be registered and config can be modified. Takes the place of the (deprecated) plugins file. [Please read the notes for examples on using this.](#setupNodeEvents) |

Reviewer: ```suggestion
| `setupNodeEvents`    | `null`                                   | Function in which node events can be registered and config can be modified. Takes the place of the (removed) plugins file. [Please read the notes for examples on using this.](#setupNodeEvents) |
```

Reviewer 3: [0ced54c](https://github.com/cypress-io/cypress-documentation/commit/0ced54cd95f1f53c9ea3b8b963748549f458b05e)

> | `supportFile`        | `cypress/support/index.js`               | Path to file to load before test files load. This file is compiled and bundled. (Pass `false` to disable)                                                                                           |

Reviewer: ```suggestion
| `supportFile`        | `cypress/support/index.js`               | Path to file to load before spec files load. This file is compiled and bundled. (Pass `false` to disable)                                                                                           |
```

> | `specPattern`        | `**/*.cy.{js,jsx,ts,tsx}`                | A String or Array of glob patterns of the test files to load. <br><br>Note that any files found matching the `e2e.specPattern` value will be automatically **excluded.**                            |

Reviewer: ```suggestion
| `specPattern`        | `**/*.cy.{js,jsx,ts,tsx}`                | A glob pattern String or Array of glob pattern Strings of the spec files to load. <br><br>Note that any files found matching the `e2e.specPattern` value will be automatically **excluded.**                            |
```

> | `excludeSpecPattern` | `['/snapshots/*', '/image_snapshots/*']` | A String or Array of glob patterns used to ignore test files that would otherwise be shown in your list of tests. [Please read the notes on using this.](#excludeSpecPattern)                       |

Reviewer: ```suggestion
| `excludeSpecPattern` | `['/snapshots/*', '/image_snapshots/*']` | A String or Array of glob patterns used to ignore spec files that would otherwise be shown in your list of specs. [Please read the notes on using this.](#excludeSpecPattern)                       |
```

Reviewer 3: [ce4d48c](https://github.com/cypress-io/cypress-documentation/commit/ce4d48cee2d5ea71ae9be4ac11f679e919987999)
